spanner/spanner_leaderboard/leaderboard.go

```diff
@@ -0,0 +1,348 @@
+	matches := regexp.MustCompile("^(.*)/databases/(.*)$").FindStringSubmatch(db)
+	if matches == nil || len(matches) != 3 {
+		return fmt.Errorf("Invalid database id %s", db)
+	}
+	op, err := adminClient.CreateDatabase(ctx, &adminpb.CreateDatabaseRequest{
+		Parent:          matches[1],
+		CreateStatement: "CREATE DATABASE `" + matches[2] + "`",
+		ExtraStatements: []string{
+			`CREATE TABLE Players(
+			    PlayerId INT64 NOT NULL,
+			    PlayerName STRING(2048) NOT NULL
+			) PRIMARY KEY(PlayerId)`,
+			`CREATE TABLE Scores(
+			    PlayerId INT64 NOT NULL,
+			    Score INT64 NOT NULL,
+			    Timestamp TIMESTAMP NOT NULL
+			    OPTIONS(allow_commit_timestamp=true)
+			) PRIMARY KEY(PlayerId, Timestamp),
+			INTERLEAVE IN PARENT Players ON DELETE NO ACTION`,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if _, err := op.Wait(ctx); err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Created database [%s]\n", db)
+	return nil
+}
+
+func insertPlayers(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	// Get number of Players
+	stmt := spanner.Statement{
+		SQL: `SELECT Count(PlayerId) as PlayerCount FROM Players`,
+	}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	row, err := iter.Next()
+	if err == iterator.Done {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	var numberOfPlayers int64
+	if err := row.Columns(&numberOfPlayers); err != nil {
+		return err
+	}
+	// Insert 100 player records into the Players table
+	for i := 1; i <= 10; i++ {
+		numberOfPlayers++
+		rand.Seed(time.Now().UnixNano())
+		min := 1000000000
+		max := 9000000000
+		PlayerId := rand.Intn(max-min) + min
+		PlayerName := fmt.Sprintf("Player %d", numberOfPlayers)
+		cols := []string{"PlayerId", "PlayerName"}
+		m := []*spanner.Mutation{
+			spanner.InsertOrUpdate("Players", cols, []interface{}{PlayerId, PlayerName}),
+		}
+		_, err := client.Apply(ctx, m)
+		if err != nil {
+			return err
+		}
+	}
+	fmt.Fprintf(w, "Inserted players \n")
+	return nil
+}
+
+func insertScores(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	var playerRecordsFound bool = false
+	// Select all player records
+	stmt := spanner.Statement{SQL: `SELECT PlayerId FROM Players`}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		playerRecordsFound = true
+		var playerID int64
+		if err := row.ColumnByName("PlayerId", &playerID); err != nil {
+			return err
+		}
+		// Intialize values for random score and date
+		rand.Seed(time.Now().UnixNano())
+		min := 1000
+		max := 1000000
+		// Insert 4 score records into the Scores table for each player in the Players table
+		for i := 1; i <= 4; i++ {
+			// Generate random score between 1,000,000 and 1,000
+			score := rand.Intn(max-min) + min
+			// Generate random day within the past two years
```

Comment:
maybe split this out into a helper function?

Reply:
Opting to leave as is for simplicity.